### ISSUE TITLE
Missing compile error for unknown name in for await loop with deconstruction

### ISSUE THREAD

**Issue by viyaha** [Microsoft/TypeScript#29797]
**TypeScript Version:**  3.4.0-dev.20190207

**Search Terms:**
for await
async
AsyncIterableIterator
AsyncIterable

**Code**

```ts
async function bar(){
    for await (const {} of imNotExisting){
        console.log("!");
    }
}

bar().catch(() => {})
```

**Expected behavior:**
TS2304 Error - Cannot find name "imNotExisting"

**Actual behavior:**
![image](https://user-images.githubusercontent.com/44086665/52414556-f2725500-2ae4-11e9-8343-3861723d90e5.png)

**Related Issues:**
Most likely same issue as #29796

**Comment by dragomirtitian** [comment]
Has nothing to do with `for await`, for any `for of` loop if the for variable is an empty binding pattern, the for expression will not get checked. So this is also not reported as an error:

```ts

function bar2(){
    for (const { } of  foo+bar){ // any expression here will not get checked 
        console.log("!");
    }
}

function bar2(){
    for (const [] of foo){
        console.log("!");
    }
}
```

This is due to the fact that the for expression is checked only as part of checking the variable declaration. If the variable declaration is an empty binding list no need arises to check the expression an so any type errors go unreported.

This behavior is not present if we use a binding list directly: 

```ts

function bar2(){
    for ({ } of foo+bar){ // error as expected
        console.log("!");
    }
}

function bar2(){
    for ([] of foo){ error as expected
        console.log("!");
    }
}
```

**Comment by RyanCavanaugh** [comment]
Definitely a duplicate of #29796 - both are symptoms of us failing to check a node
